aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -909,7 +909,7 @@ _scaled_mm_out_cuda(const Tensor& mat1, const Tensor& mat2,
 #if defined(USE_ROCM) && ROCM_VERSION >= 60000
   // rocm's hipblaslt does not yet support amax, so calculate separately
   auto out_float32 = out.to(kFloat);
-  out_float32.abs_();
+  out_float32.abs();
   amax = at::max(out_float32);
```

Comment:
Okay this makes more sense.
 What I think is happening is that for fp8_experimental we calling scaled_mm with out_dtype = fp32 (or bf16).
 This  line; `auto out_float32 = out.to(kFloat);` ends up being a no op and thus out_float32 is a view of out. we call the inplace op and this mutates the output.
 
 I think this should be:
 ` at::Tensor amax = at::max(at::abs(out));`
 because as written today the abs call will not be utilized

Reply:
That actually makes a lot of sense, since the failures were only happening with float32 as the base dtype. I'll make the change and try it out

Reply:
sounds great, could also add a regression test here: test/test_matmul_cuda.py